Router/SpecialRouter.php

```diff
@@ -29,26 +30,61 @@ class SpecialRouter extends Base
     ];
 
     /**
-     * Translate special characters in the path to names.
+     * Match a route to this router.
+     *
+     * If there is a match, replace the path with the new path.
+     * Ex. /my-module/foo-bar/baz_qux -> /my-module/foodashbar/bazunderscorequx
```

Comment:
@markshust should we use `frontName` instead of `my-module`?

Reply:
Good idea -- I'll use `front-name` though to demonstrate that the dash in the front name is not replaced.